File: LeetCode Problems/801 to 900/876. Middle of the Linked List.cpp
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* middle = head;
        int i = 0;
        while (head != nullptr) {
            i++;
            if (i % 2 == 0) {
                middle = middle->next;     
                i = 0;
            }
            head = head->next;
        }
        return middle;
    }
};

File: LeetCode Problems/801 to 900/876. Middle of the Linked List/876. Middle of the Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        int cnt = 1;
        ListNode curr = head, prev = head;
        while (curr != null) {
            if (cnt == 2) {
                prev = prev.next;
                cnt = 0;
            }
            curr = curr.next;
            cnt++;
        }
        return prev;
    }
}
